### ISSUE TITLE
security/acme-client: Support High Availability

### ISSUE THREAD

**Issue by TekunoKage** [opnsense/plugins#2950]
**Important notices**
Before you add a new report, we ask you kindly to acknowledge the following:

- [x ] I have read the contributing guide lines at https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md
- [x ] I have searched the existing issues, open and closed, and I'm convinced that mine is new.
- [x ] When the request is meant for an existing plugin, I've added its name to the title.

Is your feature request related to a problem? 
 If a firewall switchover a night before a certificate need to be updated, the next day the certificate will be expire. Currently their is no option for HA on the solution.

A simple solution will be to be able to get/update the cerificates on the "standby" that behaving as active.

An **ideal** solution will be a two way sync on this case, because if the "standby" firewall that behaving as active update the certificates, should be able to provide the updated certificate to the "primary" firewall

**Comment by fraenki** [comment]
Duplicate of https://github.com/opnsense/plugins/issues/589.
TL;DR: it would be a rather complicated feature to develeop. If someone wants to develop it: please submit a pull-request and ask for review.

>  If a firewall switchover a night before a certificate need to be updated, the next day the certificate will be expire. Currently their is no option for HA on the solution.

Slightly off-topic, but my advice would be to renew certificates more often. I think the default of most ACME-compatible Certificate Authorities is to renew 30 days in advance.

**Comment by LorenKeagle** [comment]
As opposed to fully syncing the full certificate folders, it would suffice for my needs to simply sync the configuration/automations, so that these can run independently on the backup server if necessary.  No need to transfer certificated back and forth if they can simply be regenerated on the backup machine.  Bonus if there's a setting to prevent execution based on CARP status.

Regardless of the expiration state of the certificate, it is often helpful to re-run your automations to deploy the certificate to local infrastructure.  Currently, none of that configuration is synced between nodes.

It's also a bit misleading at the moment, because from the System->High Availability->Status page, it appears that you can sync the Acme service just like everything else. But clicking the sync button spins for a while and makes it appear that it is actually doing something.  As far as I'm aware, this is the only service I have deployed that does not actually perform any settings synchronization.
